index.html

```diff
@@ -854,21 +862,39 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 '!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
 entry buffer</a>.</li>
 <li>Increase <a>resource timing buffer current size</a> by 1.</li>
+<li>Return.</li>
 </ol>
 </li>
-<li>If <a>resource timing buffer current size</a> is greater than
-or equal to <a>resource timing buffer size limit</a> and the
-<a>resource timing buffer full flag</a> is false, run the following
+<li>If <a>resource timing buffer full flag</a> is false, run the following
 substeps:
-<ol style='list-style-type: lower-latin;' data-link-for=
-"Performance">
-<li>Set the <a>resource timing buffer full flag</a> to true.</li>
+<ol style="list-style-type: lower-latin;">
+<li>Set <a>resource timing buffer full flag</a> to true.</li>
+<li>Queue a task to run <a>fire a buffer full event</a></li>
+</ol>
+<li>If <a>resource timing secondary buffer current size</a> is less than
+<a>resource timing buffer size limit</a>, add <i>new entry</i> to the
+<a>secondary entry buffer</a> and increment <a>resource timing secondary buffer
+current size</a> by 1.</li>
+</ol>
+
+<p>To <dfn>fire a buffer full event</dfn>, run the following steps:</p>
+<ol>
 <li><a data-cite="!DOM/#concept-event-fire">Fire an event</a>
 named <code>resourcetimingbufferfull</code> at the <a data-cite=
 "!HR-TIME-2/#idl-def-performance">Performance</a> object, with its
 <code>bubbles</code> attribute initialized to true.</li>
-</ol>
-</li>
+<li>If <a>resource timing buffer size limit</a> minus <a>resource timing buffer
+current size</a> is not larger than <a>resource timing secondary buffer current
+size</a>, abort these steps.</li>
```

Comment:
I think it might make sense to clear the secondary buffer even in the 'abort' case, so that we can keep listening to future entries even if the primary buffer was not cleared during this call. This is in line with the previous behavior, which would ignore entries when the buffer is full and the buffer full flag is true. Or even better but more complex, make the secondary buffer a queue, and if the queue becomes full and we want to add an entry then we pop the first in line (oldest entry). This way we maintain a list of 'recent' entries ready for when the primary buffer becomes available. WDYT? Also, I think we should add some entries from the secondary buffer even if not all of them can be added.

Reply:
I think it makes sense to clear the secondary buffer either way.
I don't know if adding more complexity here (e.g. making it a queue) would provide a lot of benefit.

I guess we can add some entries from secondary to primary even if not all of them can fit, as long as we don't trigger RTbufferfull from within "fire a buffer full event"